rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp

```diff
@@ -138,7 +138,24 @@ namespace rsx
 				glPixelStorei(GL_UNPACK_SWAP_BYTES, GL_TRUE);
 				glPixelStorei(GL_UNPACK_ROW_LENGTH, pitch / 2);
 
-				// TODO: texture swizzling
+				if (is_swizzled)
+				{
+					u16 *src, *dst;
+					u16 height = tex.height();
+					u16 width = tex.width();
+
+					unswizzledPixels = (u8*)malloc(width * height * 2);
```

Comment:
I understand. std::unique_ptr is the preferred method going forward. It would be nice to use std::unique_ptr instead of malloc, and in the rest of the given file, if you don't mind.